core/trino-main/src/main/java/io/trino/server/PluginManager.java

```diff
@@ -91,6 +91,7 @@
     private final BlockEncodingManager blockEncodingManager;
```

Comment:
Improve  the commit message. Maybe something like below?

```
Allow to inject plugin installer

That way when adding new Plugin type we can provide associated with 
plugin installer that we handle it installation without changes in PluginManager.
```

Can you please post a PR where you try to use that new concept for existing plugin types like password manager. The best would be to handle them all in one step.